src/types/enhanced_transaction_types.rs

```diff
@@ -210,6 +210,12 @@ pub struct ParseTransactionsRequest {
     pub transactions: Vec<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ParseTransactionHistoryRequest {
```

Comment:
I'd prefer if we renamed it to `ParsedTransactionHistoryRequest` so it follows the same name as its function with request added at the end